examples/vector_databases/lancedb/Youtube_QA_Bot_with_LanceDB_and_OpenAI/main.ipynb

```
@@ -0,0 +1,354 @@
+    "\n",
+    "We'll use the training split with 700 videos and 208619 sentences\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "694ceb6f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datasets import load_dataset\n",
+    "\n",
+    "data = load_dataset('jamescalam/youtube-transcriptions', split='train')\n",
+    "data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a0be4da6",
+   "metadata": {},
+   "source": [
+    "### Prepare context\n",
+    "Each item in the dataset contains just a short chunk of text. We'll need to merge a bunch of these chunks together on a rolling basis. For this demo, we'll merge 20 rows and step over 4 rows at a time. LanceDB offers chaining support so you can write declarative, readable and parameterized queries. Here we serialize to Pandas as well:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "39bbc15e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from lancedb.context import contextualize\n",
+    "\n",
+    "df = (contextualize(data.to_pandas())\n",
+    "      .groupby(\"title\").text_col(\"text\")\n",
+    "      .window(20).stride(4)\n",
+    "      .to_df())\n",
+    "df.head(1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4fe3d190",
+   "metadata": {},
+   "source": [
+    "### Create embedding function\n",
+    "To create embeddings out of the text, we'll call the OpenAI embeddings API to get embeddings. Make sure you have an API key setup and that your account has available credits."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "169134a0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openai\n",
+    "import os\n",
+    "\n",
+    "# Configuring the environment variable OPENAI_API_KEY\n",
+    "if \"OPENAI_API_KEY\" not in os.environ:\n",
+    "    # OR set the key here as a variable\n",
+    "    openai.api_key = \"sk-...\"\n",
```

We really try not to encourage hardcoding API keys in code, as it makes it very easy to accidentally leak them. Would ask you to cut this bit before we merge.

---

Thanks ted! @TevinWang can you remove the hardcoding line and just error out if the key is not found in env var?